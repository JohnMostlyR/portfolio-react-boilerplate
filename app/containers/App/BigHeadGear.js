/**
 *
 * BigHeadGear
 *
 */

import React from 'react';
import Helmet from 'react-helmet';

import { intlShape, injectIntl } from 'react-intl';
import messages from './messages';

import { config } from '../../config';

/* eslint-disable no-unused-vars */
const {
  verification: {
    bing: { siteAuth },
    facebook: { facebookAppID, facebookPagesID },
    google: { googleSiteVerification },
  },
  defaultLocale,
  locales,
  seo: { baseURL, description, openGraph, twitter },
} = config;

function constructLinks() {
  const basic = [
    {
      rel: 'canonical',
      href: baseURL,
    },
    {
      rel: 'alternate',
      href: `${baseURL}${defaultLocale}/`,
      hreflang: 'x-default',
    },
  ];

  const alternateLanguages = locales
    .map(locale => locale.toLowerCase())
    .filter(locale => locale !== defaultLocale.toLowerCase())
    .map(locale => ({
      rel: 'alternate',
      href: `${baseURL}${locale}/`,
      hreflang: locale,
    }));

  return basic.concat(alternateLanguages);
}

/**
 * @see {@link http://ogp.me/}
 * @see {@link https://developers.facebook.com/docs/sharing/webmasters}
 */
function constructOpenGraphMetaTags({ formatMessage }) {
  if (openGraph) {
    return Object.entries(openGraph)
      .filter(([, { content }]) => content)
      .reduce((previous, [property, { content, isMessageId }]) => {
        if (property.toLowerCase() === 'image' && typeof content === 'object') {
          const IMG = Object.entries(content).map(
            ([imageTag, imageTagValue]) => {
              if (imageTag.toLowerCase() === 'url') {
                return { property: 'og:image', content: imageTagValue };
              }

              return {
                property: `og:image:${imageTag}`,
                content: imageTagValue,
              };
            },
          );

          return previous.concat(...IMG);
        }

        if (isMessageId) {
          return previous.concat({
            property: `og:${property}`,
            content: formatMessage({ ...messages[content] }),
          });
        }

        return previous.concat({ property: `og:${property}`, content });
      }, []);
  }

  return [];
}

/**
 * @see {@link https://developer.twitter.com/en/docs/tweets/optimize-with-cards/overview/summary}
 */
function constructTwitterMetaTags({ formatMessage }) {
  if (twitter) {
    return Object.entries(twitter)
      .filter(([, { content }]) => content)
      .map(([property, { content, isMessageId }]) => {
        if (isMessageId) {
          return {
            property: `twitter:${property}`,
            content: formatMessage({ ...messages[content] }),
          };
        }

        return { property: `twitter:${property}`, content };
      });
  }

  return [];
}

function constructBasicMetaTags({ formatMessage }) {
  const { content, isMessageId } = description;

  if (isMessageId) {
    return [
      {
        name: 'description',
        content: formatMessage({ ...messages[content] }),
      },
    ];
  }

  return [
    {
      name: 'description',
      content,
    },
  ];
}

function constructMetaTags({ formatMessage }) {
  const facebookTags = [];

  if (facebookAppID) {
    facebookTags.push({
      property: 'fb:app_id',
      content: facebookAppID,
    });
  }

  return [].concat(
    constructBasicMetaTags({ formatMessage }),
    constructOpenGraphMetaTags({ formatMessage }),
    constructTwitterMetaTags({ formatMessage }),
    facebookTags,
  );
}

function BigHeadGear({ intl: { locale, formatMessage } }) {
  return (
    <Helmet
      htmlAttributes={{ lang: locale }}
      title={formatMessage({ ...messages.title })}
      meta={constructMetaTags({ formatMessage })}
    />
  );
}

BigHeadGear.propTypes = {
  intl: intlShape.isRequired,
};

export default injectIntl(BigHeadGear);
