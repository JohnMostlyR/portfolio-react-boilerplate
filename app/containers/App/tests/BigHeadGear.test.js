import React from 'react';

import BigHeadGear from '../BigHeadGear';

import { shallowWithIntl } from '../../../helpers/intl-enzyme-test-helper';

describe('<BigHeadGear />', () => {
  it('should render and match the snapshot', () => {
    const renderedComponent = shallowWithIntl(<BigHeadGear />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
