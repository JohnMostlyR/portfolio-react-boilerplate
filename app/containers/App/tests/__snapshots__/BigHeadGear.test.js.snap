// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BigHeadGear /> should render and match the snapshot 1`] = `
<BigHeadGear
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {
        "portfolio.component.BrandLink.title": "← Back to the homepage",
        "portfolio.component.ContentLoadingIndicator.error": "Terribly sorry, but the content cannot be loaded at this time",
        "portfolio.component.ContentLoadingIndicator.pastDelay": "Loading content",
        "portfolio.component.ContentLoadingIndicator.timedOut": "Loading is taking longer than expected",
        "portfolio.components.Form.helperText.range": "Should be between {minLength, number} and {maxLength, number} characters. Is now: {count, number}",
        "portfolio.components.Form.helperText.valid": "Thanks!",
        "portfolio.components.LanguageMenu.buttonLabel": "Language",
        "portfolio.components.LanguageMenu.label": "Choose a Language",
        "portfolio.components.LanguageMenu.title": "Language menu",
        "portfolio.components.Project.detailsLink": "More details →",
        "portfolio.components.ProjectDetails.backLink": "← Back to Projects",
        "portfolio.components.SubmitButton.error": "Failed. Please try again",
        "portfolio.components.SubmitButton.idle": "Send",
        "portfolio.components.SubmitButton.sending": "Sending...",
        "portfolio.components.SubmitButton.success": "Sent! I'll respond as soon as possible",
        "portfolio.containers.ContactPage.emailError": "Your email address is required.",
        "portfolio.containers.ContactPage.emailLabel": "Your email address",
        "portfolio.containers.ContactPage.emailPlaceholder": "(e.g. my@email.com)",
        "portfolio.containers.ContactPage.emailText": "Please add your email address here.",
        "portfolio.containers.ContactPage.meta.description": "For contacting me, please fill in this form. I will respond as soon as possible",
        "portfolio.containers.ContactPage.nameError": "Your name is required.",
        "portfolio.containers.ContactPage.nameLabel": "Your name",
        "portfolio.containers.ContactPage.namePlaceholder": "(e.g. John Doo)",
        "portfolio.containers.ContactPage.nameText": "Please add your name here.",
        "portfolio.containers.ContactPage.pageTitle": "CONTACT FORM",
        "portfolio.containers.ContactPage.requirementOne": "Please fill in all fields",
        "portfolio.containers.ContactPage.subjectError": "A subject is required.",
        "portfolio.containers.ContactPage.subjectLabel": "Subject",
        "portfolio.containers.ContactPage.subjectText": "Please provide a subject for your message.",
        "portfolio.containers.ContactPage.title": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
        "portfolio.containers.ContactPage.yourMessageError": "A message is required.",
        "portfolio.containers.ContactPage.yourMessageLabel": "Your message",
        "portfolio.containers.ContactPage.yourMessageText": "Please add your message here.",
        "portfolio.meta.description": "This is the portfolio website of Johan Meester. An Amsterdam based Front-End Developer specialized in React JS",
        "portfolio.meta.siteName": "Johan Meester : Portfolio",
        "portfolio.nav.about": "About",
        "portfolio.nav.contact": "Contact",
        "portfolio.nav.header": "Site navigation",
        "portfolio.nav.home": "Intro",
        "portfolio.nav.projects": "Projects",
        "portfolio.nav.showMenuButton": "MENU",
        "portfolio.nav.skills": "Skills",
        "portfolio.noscript": "If you are seeing this message, that means that JavaScript has been disabled on your browser. Please enable JavaScript to make this app work.",
        "portfolio.page.AboutPage.meta.description": "A short description about me and my professional life",
        "portfolio.page.AboutPage.pageTitle": "ABOUT ME",
        "portfolio.page.AboutPage.title": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
        "portfolio.page.NotFoundPage.contact": "contact",
        "portfolio.page.NotFoundPage.content": "I am sorry, but the page you requested does not exist or has been removed.

Please go to the {home} page or, if you think this is an error, to the {contact} page and contact me.",
        "portfolio.page.NotFoundPage.home": "start",
        "portfolio.page.NotFoundPage.pageDescription": "The page you requested does not exist or has been removed. Sorry about that",
        "portfolio.page.NotFoundPage.pageTitle": "NOT FOUND",
        "portfolio.page.NotFoundPage.title": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
        "portfolio.page.ProjectsPage.meta.description": "This is an overview of my projects. Projects are sorted by date with the most recent on top",
        "portfolio.page.ProjectsPage.pageTitle": "MY PROJECTS",
        "portfolio.page.ProjectsPage.title": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
        "portfolio.page.SkillsPage.meta.description": "A brief overview of my professional skills",
        "portfolio.page.SkillsPage.pageTitle": "MY SKILLS",
        "portfolio.page.SkillsPage.title": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
        "portfolio.page.index.header.callToAction": "CONTACT ME",
        "portfolio.page.index.header.greeting": "WELCOME",
        "portfolio.page.index.header.intro": "I am {name}",
        "portfolio.page.index.header.look": "HAVE A LOOK AROUND",
        "portfolio.page.index.header.trade": "FRONT-END DEVELOPER",
        "portfolio.page.index.meta.description": "Welcome to the portfolio of Johan Meester, Front-End Developer",
        "portfolio.page.index.pageTitle": "INTRODUCTION",
        "portfolio.page.index.title": "Johan Meester : Portfolio : Front-End Developer : {pageTitle}",
        "portfolio.skiplink": "skip to main content",
        "portfolio.social_links.header": "Also find me on:",
        "portfolio.title": "Johan Meester : Portfolio : Front-End Developer",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
/>
`;
